client/src/components/BlockEditor/BlockEditorEditing.tsx

```diff
@@ -202,6 +318,52 @@ export class BlockEditorEditingPresentational extends React.Component<
   };
 
   private onChangeCallback = (c: any) => {
+    // first check to see if document changed, which we can do with !== b/c
+    // Slate uses immutable objects, if it has changed then normalize wrt
+    // pointer spacing
+    if (c.value.document !== this.props.block.value.document) {
+      c = this.normalizeChangeWrtTextNodeSpacing(c);
+    }
+
+    // the rest of this method before the last two lines are trying to prevent
+    // the user from using a click top get the text cursor in off limit spacing
+    // the onClick handler didn't seem to have access to the new cursor position
+    // so this logic lives here instead
+
+    const value = c.value;
+    const selection = value.selection;
+
+    const anchorKey = selection.anchorKey;
+    const focusKey = selection.focusKey;
+    const anchorOffset = selection.anchorOffset;
+    const focusOffset = selection.focusOffset;
+
+    const selectionIsExpanded = (anchorKey !== focusKey) || (anchorOffset !== focusOffset);
+
+    // if selection is expanded, then we don't do this
+    // this allows mouse dragging to select across pointers
```

Comment:
The code below is somewhat difficult to read without knowing Slate. I wonder if there are abstractions that would help. It's also plausible that the behavior in the `!selectionIsExpanded` case should be factored out.

Reply:
Not exactly what you were suggesting, but I factored out the logic for determining whether a selection is expanded or not: https://github.com/oughtinc/mosaic/pull/145/commits/73b23db2e2fd114e3387e7516f6b340cf8826eb9